### ISSUE TITLE
Rule Request: Implicit initialization of optional bindings

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#1940]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

> 1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Swift supports both default implicit initialization of optional bindings and explicit.

These are equivalent:

```swift
struct A {
  var x: Int?
}

struct A {
  var x: Int? = nil
}
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

I think you should be able to enforce it either way, so either `"always"` or `"never"` implicit init of optional bindings.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

For `"always"`, this would trigger a violation:

```swift
struct A {
  var x: Int? = nil
}
```

For `"never"`, this would trigger a violation:

```swift
struct A {
  var x: Int?
}
```

> 4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

Opt-in as it's opinionated.

**Comment by jpsim** [comment]
Note that the "never" case already exists as the [Redundant Optional Initialization Rule](https://github.com/realm/SwiftLint/blob/master/Rules.md#redundant-optional-initialization).

I'm ok if we want to rename this rule and make it configurable to both cases.

**Comment by leo-lem** [comment]
can i work on this? :)

**Comment by SimplyDanny** [comment]
> can i work on this? :)

Sure. Please go ahead. Nobody has worked on it for the last 8 years, hence double work is pretty unlikely. 😉
